SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h

```diff
@@ -2,11 +2,49 @@
 
 #pragma once
 
+#include "Utils/SpatialBasicAwaiter.h"
 #include <WorkerSDK/improbable/c_worker.h>
+
 #include "SpatialWorkerFlags.generated.h"
 
-DECLARE_DYNAMIC_DELEGATE_TwoParams(FOnWorkerFlagsUpdatedBP, const FString&, FlagName, const FString&, FlagValue);
-DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnWorkerFlagsUpdated, const FString&, FlagName, const FString&, FlagValue);
+DECLARE_DYNAMIC_DELEGATE_TwoParams(FOnAnyWorkerFlagUpdatedBP, const FString&, FlagName, const FString&, FlagValue);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnAnyWorkerFlagUpdated, const FString&, FlagName, const FString&, FlagValue);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FOnWorkerFlagUpdatedBP, const FString&, FlagValue);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnWorkerFlagUpdated, const FString&, FlagValue);
+
+UENUM()
+enum class ESpatialCallbackStyle : uint8
+{
+	InvokeImmidiatelyIfAlreadySet,
+	InvokeOnNewUpdateOnly
+};
+
+USTRUCT()
```

Comment:
Why does this need to be a UStruct?

Reply:
Contains a UObject -> USpatialBasicAwaiter* FlagReady;

Reply:
I am pretty sure I remember UPROPERTY being required for GC to work, but its been a while so maybe I am misrembering.

Reply:
It does but I don't see why we need the awaiter, or why that needs to be a UObject.